src/redux/books/books.js

```diff
@@ -1,42 +1,81 @@
 // Define action types for adding and removing a book.
+import { createAsyncThunk } from '@reduxjs/toolkit';
+
 const ADD_BOOK = 'bookstore / books / ADD_BOOK';
 const REMOVE_BOOK = 'bookstore / books / REMOVE_BOOK';
-const initialState = [
-  {
-    id: 1,
-    title: 'THE HUNGER GAMES',
-    author: 'Suzanne Collins',
-  },
-  {
-    id: 2,
-    title: 'DUNE',
-    author: 'Frank Herbert',
-  },
-  {
-    id: 3,
-    title: 'Capital in the Twenty-first Century',
-    author: 'Suzanne Collins',
-  },
-];
+// const APP_ID = 'WLWRGsXNCux9kjlYUIID';
+const FETCH_BOOKS = 'bookstore/books/FETCH_BOOKS';
+
 // Action Creators
-export const addBook = (book) => ({
-  type: ADD_BOOK,
-  payload: book,
+const addBook = createAsyncThunk(ADD_BOOK, async (book, { dispatch }) => {
+  await fetch('https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/WLWRGsXNCux9kjlYUIID/books/', {
+    method: 'POST',
+    headers: {
+      Accept: 'application/json',
+      'Content-Type': 'application/json',
+      'Access-Control-Allow-Origin': '*',
+    },
+    body: JSON.stringify(book),
+  });
+  dispatch({
+    type: ADD_BOOK,
+    book,
+  });
+});
+const fetchBooks = createAsyncThunk(FETCH_BOOKS, async (args, { dispatch }) => {
+  const response = await fetch('https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/WLWRGsXNCux9kjlYUIID/books/', {
+    method: 'GET',
+  });
+  const data = await response.json();
+  const bookIDs = Object.entries(data);
+  const books = [];
+  bookIDs.forEach(([key, book]) => {
+    const id = key;
+    const { title, author, category } = book[0];
+    books.push({
+      id, title, author, category,
+    });
+  });
+  dispatch({
+    type: FETCH_BOOKS,
+    payload: books,
+  });
+  // console.log(books);
```

Comment:
- [ ] kindly consider removing all commented code from your repo, it makes your code looks messy and hard to maintain